File: Source/StringRelatedOperations/replace.c
#include <stdlib.h>
#include <string.h>

char* replace(char* str, char oldChar, char newChar) {
    int len = strlen(str);
    char* result = (char*)malloc(sizeof(char) * (len + 1));
    int i;

    for(i = 0; i < len; i++) {
        if(str[i] == oldChar) {
            result[i] = newChar;
        } else {
            result[i] = str[i];
        }
    }
    result[i] = '\0';

    return result;
}

File: Source/StringRelatedOperations/replace.js
function replace(str, oldChar, newChar) {
    var result = "";
    for(var i = 0; i < str.length; i++) {
        if(str[i] == oldChar) {
            result += newChar;
        } else {
            result += str[i];
        }
    }
    return result;
}

var str = "Hello, World!";
var replaced = replace(str, 'o', 'a');

console.log(replaced);
